src/bun.js/events.exports.js

```diff
@@ -0,0 +1,464 @@
+  }
+  return this;
+};
+
+function overflowWarning(emitter, type, handlers) {
+  handlers.warned = true;
+  const warn = new Error(
+    `Possible EventEmitter memory leak detected. ${handlers.length} ${String(type)} listeners ` +
+      `added to [${emitter.constructor.name}]. Use emitter.setMaxListeners() to increase limit`,
+  );
+  warn.name = "MaxListenersExceededWarning";
+  warn.emitter = emitter;
+  warn.type = type;
+  warn.count = handlers.length;
+  process.emitWarning(warn);
+}
+
+function onceWrapper(type, listener, ...args) {
+  this.removeListener(type, listener);
+  listener.apply(this, args);
+}
+
+EventEmitter.prototype.once = function once(type, fn) {
+  checkListener(fn);
+  const bound = onceWrapper.bind(this, type, fn);
+  bound.listener = fn;
+  this.addListener(type, bound);
+  return this;
+};
+
+EventEmitter.prototype.prependOnceListener = function prependOnceListener(type, fn) {
+  checkListener(fn);
+  const bound = onceWrapper.bind(this, type, fn);
+  bound.listener = fn;
+  this.prependListener(type, bound);
+  return this;
+};
+
+EventEmitter.prototype.removeListener = function removeListener(type, fn) {
+  checkListener(fn);
+  var { _events: events } = this;
+  if (!events) return this;
+  var handlers = events[type];
+  if (!handlers) return this;
+  var length = handlers.length;
+  let position = -1;
+  for (let i = length - 1; i >= 0; i--) {
+    if (handlers[i] === fn || handlers[i].listener === fn) {
+      position = i;
+      break;
+    }
+  }
+  if (position < 0) return this;
+  if (position === 0) {
+    handlers.shift();
+  } else {
+    handlers.splice(position, 1);
+  }
+  if (handlers.length === 0) {
+    delete events[type];
+    this._eventsCount--;
+  }
+  return this;
+};
+
+EventEmitter.prototype.off = EventEmitter.prototype.removeListener;
+
+EventEmitter.prototype.removeAllListeners = function removeAllListeners(type) {
+  var { _events: events } = this;
+  if (type && events) {
+    if (events[type]) {
+      delete events[type];
+      this._eventsCount--;
+    }
+  } else {
+    this._events = { __proto__: null };
+  }
+  return this;
+};
+
+EventEmitter.prototype.listeners = function listeners(type) {
+  var { _events: events } = this;
+  if (!events) return [];
+  var handlers = events[type];
+  if (!handlers) return [];
+  return handlers.map(x => x.listener ?? x);
+};
+
+EventEmitter.prototype.rawListeners = function rawListeners(type) {
+  var { _events } = this;
+  if (!_events) return [];
+  var handlers = _events[type];
+  if (!handlers) return [];
+  return handlers.slice();
+};
+
+EventEmitter.prototype.listenerCount = function listenerCount(type) {
+  var { _events: events } = this;
+  if (!events) return 0;
+  return events[type]?.length ?? 0;
+};
+
+EventEmitter.prototype.eventNames = function eventNames() {
+  return this._eventsCount > 0 ? Reflect.ownKeys(this._events) : [];
+};
+
+EventEmitter.prototype[kCapture] = false;
+
+function once(emitter, type, { signal } = {}) {
```

Comment:
we can't set a default argument in the function this way

people will pass `null` or something truthy and then get a very strange error